src/Core/Context/LanguageContextBuilder.php

```diff
@@ -121,5 +126,6 @@ public function buildLegacyContext(): void
         $this->assertArguments();
         $language = $this->objectModelLanguageRepository->get(new LanguageId($this->languageId));
         $this->contextStateManager->setLanguage($language);
+        $this->contextStateManager->setCurrentLocale($this->getLocaleByLanguage($language));
```

Comment:
All good but we need to update the related tests in `Tests\Unit\Core\Context\LanguageContextBuilderTest::testBuildLegacyContext`

We need to assert that when `buildLegacyContext` is called the ContextStateManager is also called to override the locale